setup.py

```diff
@@ -165,466 +26,206 @@ def _get_interop_filename():
-
-    def _find_msbuild_tool_15(self):
-        """Return full path to one of the Microsoft build tools"""
-        try:
-            basePaths = subprocess.check_output(
-                [
-                    "tools\\vswhere\\vswhere.exe",
-                    "-latest",
-                    "-version",
-                    "[15.0,)",
-                    "-requires",
-                    "Microsoft.Component.MSBuild",
-                    "-find",
-                    "MSBuild\**\Bin\MSBuild.exe",
-                ]
-            ).splitlines()
-            if len(basePaths):
-                return basePaths[0].decode(sys.stdout.encoding or "utf-8")
-            else:
-                raise RuntimeError("MSBuild >=15.0 could not be found.")
-        except subprocess.CalledProcessError as e:
-            raise RuntimeError(
-                "MSBuild >=15.0 could not be found. {0}".format(e.output)
-            )
 
+class BuildDotnet(build_ext):
+    """Build command for dotnet-cli based builds"""
 
-class InstallLibPythonnet(install_lib.install_lib):
-    def install(self):
-        if not os.path.isdir(self.build_dir):
-            self.warn(
-                "'{0}' does not exist -- no Python modules"
-                " to install".format(self.build_dir)
-            )
-            return
+    description = "Build DLLs with dotnet-cli"
+    user_options = [("dotnet-config", None, "dotnet build configuration")]
 
-        if not os.path.exists(self.install_dir):
-            self.mkpath(self.install_dir)
+    def initialize_options(self):
+        self.dotnet_config = "release"
+        super().initialize_options()
 
-        # only copy clr.pyd/.so
-        for srcfile in glob.glob(os.path.join(self.build_dir, "clr.*")):
-            destfile = os.path.join(self.install_dir, os.path.basename(srcfile))
-            self.copy_file(srcfile, destfile)
+    def finalize_options(self):
+        super().finalize_options()
 
+    def get_source_files(self):
+        return super().get_source_files()
 
-class InstallDataPythonnet(install_data.install_data):
     def run(self):
-        build_cmd = self.get_finalized_command("build_ext")
-        install_cmd = self.get_finalized_command("install")
-        build_lib = os.path.abspath(build_cmd.build_lib)
-        install_platlib = os.path.relpath(install_cmd.install_platlib, self.install_dir)
-
-        for i, data_files in enumerate(self.data_files):
-            if isinstance(data_files, str):
-                self.data_files[i] = data_files[i].format(build_lib=build_lib)
+        orig_modules = self.distribution.ext_modules
+        dotnet_modules = [lib for lib in orig_modules if isinstance(lib, DotnetLib)]
+        other_modules = [lib for lib in orig_modules if not isinstance(lib, DotnetLib)]
+
+        if dotnet_modules:
+            if os.path.isfile(CONFIGURED_PROPS):
+                self.announce("Already configured", level=distutils.log.INFO)
```

Comment:
Do these ever run in parallel? Couldn't there be a race condition around existence of this file?

Reply:
Not really, but as said, I'll just make these always run. The time to do this is completely dwarfed by the actual calls to `dotnet build`.